forc-plugins/forc-crypto/src/keys/vanity.rs

```diff
@@ -0,0 +1,440 @@
+        let private_key =
+            SecretKey::new_from_mnemonic_phrase_with_path(&mnemonic, &derivation_path)?;
+        (private_key, Some(mnemonic))
+    } else {
+        (SecretKey::random(&mut rng), None)
+    };
+
+    let public = PublicKey::from(&private_key);
+    let hashed = public.hash();
+    let address = Bech32Address::new(FUEL_BECH32_HRP, hashed);
+
+    Ok((address.into(), private_key, mnemonic))
+}
+
+pub trait VanityMatcher: Send + Sync {
+    fn is_match(&self, addr: &Address) -> bool;
+}
+
+struct HexMatcher {
+    left: Vec<u8>,
+    right: Vec<u8>,
+}
+
+impl VanityMatcher for HexMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let bytes = addr.as_ref();
+        bytes.starts_with(&self.left)
+            && bytes[bytes.len() - self.right.len()..]
+                .iter()
+                .zip(self.right.iter())
+                .all(|(a, b)| a.eq_ignore_ascii_case(b))
+    }
+}
+
+struct LeftHexMatcher {
+    left: Vec<u8>,
+}
+
+impl VanityMatcher for LeftHexMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let bytes = addr.as_ref();
+        bytes.starts_with(&self.left)
+    }
+}
+
+struct RightHexMatcher {
+    right: Vec<u8>,
+}
+
+impl VanityMatcher for RightHexMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let bytes = addr.as_ref();
+        bytes[bytes.len() - self.right.len()..]
+            .iter()
+            .zip(self.right.iter())
+            .all(|(a, b)| a.eq_ignore_ascii_case(b))
+    }
+}
+
+struct LeftExactRightRegexMatcher {
+    left: Vec<u8>,
+    right: Regex,
+}
+
+impl VanityMatcher for LeftExactRightRegexMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let bytes = addr.as_ref();
+        bytes.starts_with(&self.left) && self.right.is_match(&hex::encode(bytes))
+    }
+}
+
+struct LeftRegexRightExactMatcher {
+    left: Regex,
+    right: Vec<u8>,
+}
+
+impl VanityMatcher for LeftRegexRightExactMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let bytes = addr.as_ref();
+        self.left.is_match(&hex::encode(bytes))
+            && bytes[bytes.len() - self.right.len()..]
+                .iter()
+                .zip(self.right.iter())
+                .all(|(a, b)| a.eq_ignore_ascii_case(b))
+    }
+}
+
+struct SingleRegexMatcher {
+    re: Regex,
+}
+
+impl VanityMatcher for SingleRegexMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let addr = hex::encode(addr.as_ref());
+        self.re.is_match(&addr)
+    }
+}
+
+struct RegexMatcher {
+    left: Regex,
+    right: Regex,
+}
+
+impl VanityMatcher for RegexMatcher {
+    fn is_match(&self, addr: &Address) -> bool {
+        let addr = hex::encode(addr.as_ref());
+        self.left.is_match(&addr) && self.right.is_match(&addr)
+    }
+}
+
+fn parse_pattern(pattern: &str, is_start: bool) -> Result<Either<Vec<u8>, Regex>> {
+    if let Ok(decoded) = hex::decode(pattern) {
```

Comment:
Perhaps we could validate the the pattern length is no longer than 64

```rust
if pattern.len() > 64 { 
    return Err(anyhow!("Pattern too long: max 64 characters"));
}
````

Reply:
addressed: https://github.com/FuelLabs/sway/pull/6661/commits/b16c926bada977a31c774bbd9197f969f3e8c516